staker-rewards/test/requirements.test.js

```diff
@@ -0,0 +1,122 @@
+import { OperatorAuthorizations } from "../lib/requirements.js"
+
+const { expect } = chai
+
+const setupContext = (context) => {
+  context.contracts.BondedECDSAKeepFactory = {
+    deployed: () => ({
+      methods: {
+        getSortitionPool: (application) => ({
+          call: async (_, blockNumber) => {
+            return SortitionPoolAddress
+          },
+        }),
+        isOperatorAuthorized: (operator) => ({
+          call: mockMethod(
+            BondedECDSAKeepFactoryAddress,
+            "isOperatorAuthorized",
+            (inputs) => inputs.operator === operator,
+            false
+          ),
+        }),
+      },
+    }),
+  }
+
+  context.contracts.KeepBonding = {
+    deployed: () => ({
+      options: {
+        address: KeepBondingAddress,
+      },
+      methods: {
+        hasSecondaryAuthorization: (operator, sortitionPool) => ({
+          call: mockMethod(
+            KeepBondingAddress,
+            "hasSecondaryAuthorization",
+            (inputs) =>
+              inputs.operator === operator &&
+              inputs.poolAddress === sortitionPool,
+            false
+          ),
+        }),
+      },
+    }),
+  }
+
+  return context
+}
+
+describe("requirements", async () => {
+  const interval = { startBlock: 1000, endBlock: 2000 }
+
+  let mockContext
+  let requirements
+
+  before(() => {
+    mockContext = createMockContext()
+    setupContext(mockContext)
+  })
+
+  beforeEach(async () => {
+    requirements = await Requirements.initialize(mockContext, interval)
+  })
+
+  describe("deauthorizations check", async () => {
+    it("finds operators from cached transactions for the current interval", async () => {
+      // The mocked transaction cache contains following deauthoriation transactions:
+      //   1. made before interval start for the sortition pool
+      //   2. made on interval start for the sortition pool
+      //   3. made during interval for another sortition pool
+      //   4. made during interval for the sortition pool
+      //   5. made on interval end for the sortition pool
+      //   6. made after interval end for the sortition pool
+      //
+      // We assume the check to find operators 2 and 4.
+
+      await requirements.checkDeauthorizations()
+
+      expect(requirements.operatorsDeauthorizedInInterval).to.have.members([
+        "0xa000000000000000000000000000000000000002",
+        "0xa000000000000000000000000000000000000004",
+      ])
+    })
+  })
+
+  describe("operator's authorizations check", async () => {
+    beforeEach(async () => {
+      await requirements.checkDeauthorizations()
+    })
+
+    it("discovers authorizations on interval start", async () => {
+      const operator = "0xA000000000000000000000000000000000000001"
+
+      const result = await requirements.checkAuthorizations(operator)
+
+      expect(result).deep.equal(
+        new OperatorAuthorizations(operator, true, true, false)
+      )
+    })
```

Comment:
Ideally, we should expand tests to explicitly cover:
- no factory authorized on interval start
- no sortition pool authorized on interval start

Reply:
LOL :) Yeah, they are reading values from the mock but at least you know that your code is executing the required checks that go to the source of truth (database/cache/chain) instead of doing, for example:
```
if (wasSortitionPoolDeauthorized(operator)) {
  return (true, false, true);
} else {
  return (true, true, false);
}
```

Reply:
I've added a test case. Would it be enough or do you think we should add more?

Reply:
We are good.